hs_core/tasks.py

```diff
@@ -166,6 +167,15 @@ def create_bag_by_irods(resource_id, istorage=None):
     from hs_core.hydroshare.utils import get_resource_by_shortkey
 
     res = get_resource_by_shortkey(resource_id)
+    try:
+        # we will create xml metadata files every time prior to creating the bag
+        # the assumption here is that bag is being created due to the resource
+        # modified (AVU)
+        create_bag_files(res, fed_zone_home_path=res.resource_federation_path)
+    except Exception as ex:
+        logger.error('Failed to create bag files. Error:{}'.format(ex.message))
+        return False
+

```

Comment:
@pkdash What about metadata is not dirty when the bag is downloaded? Addition of create_bag_files() call here is not needed in this case. So a condition needs to be added here to check metadata_dirty flag, and only call create_bag_files() when metadata_dirty is true.